sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -207,3 +207,47 @@ def test_mutual_info_options(global_dtype):
         assert_allclose(mi_5, mi_6)
 
         assert not np.allclose(mi_1, mi_3)
+
+
+def test_mutual_information_symmetry_classif_regression(global_random_seed):
+    """Check that `mutual_info_classif` and `mutual_info_regression` are
+    symmetric by switching the target `y` as `feature` in `X` and vice
+    versa.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/23720
+    """
+    rng = np.random.RandomState(global_random_seed)
+    n = 100
+    d, c = rng.randint(10, size=n), rng.normal(0, 1, size=n)
+
+    mi_classif = mutual_info_classif(
+        c[:, None], d, discrete_features=[False], random_state=123
+    )
+
+    mi_regression = mutual_info_regression(
+        d[:, None], c, discrete_features=[True], random_state=123
+    )
+
+    assert mi_classif == pytest.approx(mi_regression)
+
+
+def test_mutual_info_symmetry_classif_regression_correlated():
```

Comment:
Is this a repeat of `test_mutual_information_symmetry_classif_regression`? Also, isn't `test_mutual_information_symmetry_classif_regression` a stronger test since `c` is not the same as `d`?

Reply:
I suggested/asked if it was worth testing that the MI is symmetric when `c` and `d` are independent (`test_mutual_information_symmetry_classif_regression`) and again a case where the MI should be high(er) for `c` and `d` that are correlated/not independent.

So they are kinda repeats of each other, but also not. I don't know if you need both or just one :-/

Reply:
Ah I see. I'm okay with keeping both tests. In that case, we can parameterize over the two cases:

```python
@pytest.mark.parametrize("correlated", [True, False])
def test_mutual_information_symmetry_classif_regression(global_random_seed, correlated):
    ...
    d = rng.randint(10, size=n)
    if correlated:
        c = d.astype(np.float64)
    else:
        c = rng.normal(0, 1, size=n)
```